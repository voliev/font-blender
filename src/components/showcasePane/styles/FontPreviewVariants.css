.FontPreviewVariants .variant {
  align-items: center;
  display: flex;
}

.FontPreviewVariants .variant-value {
  flex: 0 0 7rem;
  margin-right: var(--space-md);
}

.FontPreviewVariants .variant-example {
  flex: 0 auto auto;
  margin: 0;
}
