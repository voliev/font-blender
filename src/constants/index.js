// *** ACTION TYPES ***

// Background
export const SET_BACKGROUND_COLOR = "SET_BACKGROUND_COLOR";

// Filteres
export const SET_CATEGORIES_FILTER = "SET_CATEGORIES_FILTER";
export const SET_SORTING_ORDER = "SET_SORTING_ORDER";
export const SET_SORTING_PARAM = "SET_SORTING_PARAM";
export const SET_TITLE_FILTER = "SET_TITLE_FILTER";

// Fonts
export const FETCH_FONTS_DATA = "FETCH_FONTS_DATA";
export const FETCH_FONTS_DATA_SUCCESS = "FETCH_FONTS_DATA_SUCCESS";
export const FETCH_FONTS_DATA_FAILURE = "FETCH_FONTS_DATA_FAILURE";
export const SET_FONTS = "SET_FONTS";
export const UPDATE_FONT = "UPDATE_FONT";

// Notifications
export const ADD_NOTIFICATION = "ADD_NOTIFICATION";
export const REMOVE_NOTIFICATION = "REMOVE_NOTIFICATION";

// Showcase
export const DISPLAY_SHOWCASE_TEXT = "DISPLAY_SHOWCASE_TEXT";
export const DISPLAY_STYLES = "DISPLAY_STYLES";
export const DISPLAY_SHOWCASE_FONT_PREVIEW = "DISPLAY_SHOWCASE_FONT_PREVIEW";

// Styles
export const SET_DEFAULT_STATE = "SET_DEFAULT_STATE";
export const UNDO_RESET = "UNDO_RESET";

// Text styles
export const SET_STYLES_INHERITANCE = "SET_STYLES_INHERITANCE";
export const UPDATE_ELEMENT_STYLES = "UPDATE_ELEMENT_STYLES";

// Selected fonts
export const ADD_SELECTED_FONT = "ADD_SELECTED_FONT";
export const CLEAR_SELECTED_FONTS = "CLEAR_SELECTED_FONTS";
export const REMOVE_FONT_FROM_SELECTED = "REMOVE_FONT_FROM_SELECTED";
export const UNDO_DESELECTED = "UNDO_DESELECTED";

// Type scale
export const SET_DEFAULT_TYPE_SCALE_STATE = "SET_DEFAULT_TYPE_SCALE_STATE";
export const SET_FONT_SIZE = "SET_FONT_SIZE";
export const SET_TYPE_SCALE = "SET_TYPE_SCALE";

// *** MISCELLANEOUS ***

// Google Fonts API endpoint URL
export const GOOGLE_FONTS_API_ENDPOINT =
  "https://www.googleapis.com/webfonts/v1/webfonts";
// Fonts' bundle size that should be prefetched
export const BUNDLE_SIZE = 50;
// Available text elements for styling
export const ELEMENTS = ["body", "h1", "h2", "h3", "h4", "li", "small"];
// Font's character range that needs to be prefetched
export const CHAR_RANGE =
  "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
// Font showcase string
export const SHOWCASE_STRING =
  "That's the beauty of music. They can't get that from you...";
// Default showcase font size
export const DEFAULT_FONT_SIZE = 16;
// Available type scale values
export const SCALE_VALUES = {
  1.067: "Minor Second",
  1.125: "Major Second",
  1.2: "Minor Third",
  1.25: "Major Third",
  1.333: "Perfect Fourth",
  1.414: "Augmented Fourth",
  1.5: "Perfect Fifth",
  1.618: "Golden Ratio"
};
// Minimum gap between the color picker icon and viewport
// edge required to place the color picker
export const MIN_GAP = 300;
// Notification snackbar visibility timeout
export const TIMEOUT = 10000;
