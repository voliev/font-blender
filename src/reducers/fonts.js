import { combineReducers } from "redux";
import {
  FETCH_FONTS_DATA,
  FETCH_FONTS_DATA_FAILURE,
  FETCH_FONTS_DATA_SUCCESS,
  SET_FONTS,
  UPDATE_FONT
} from "../constants";

export const fontsDataStatus = (state = "", action) => {
  switch (action.type) {
    case FETCH_FONTS_DATA:
      return "fetching";
    case FETCH_FONTS_DATA_SUCCESS:
      return "success";
    case FETCH_FONTS_DATA_FAILURE:
      return "failure";
    default:
      return state;
  }
};

const initialState = {};

export const all = (state = initialState, action) => {
  switch (action.type) {
    case SET_FONTS:
      return action.fonts;
    case UPDATE_FONT:
      return {
        ...state,
        [action.family]: {
          ...state[action.family],
          ...action.updates
        }
      };
    default:
      return state;
  }
};

export default combineReducers({
  all,
  fontsDataStatus
});

export const getFonts = state => state.fonts.all;
export const getFont = (state, family) => state.fonts.all[family];
export const getFontsDataStatus = state => state.fonts.fontsDataStatus;
