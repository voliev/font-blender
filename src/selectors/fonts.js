import isEqual from "lodash.isequal";

export default function() {
  const memo = {
    filters: {},
    result: []
  };

  return function(fonts, filters) {
    const families = Object.keys(fonts);

    if (families.length === 0) {
      memo.filters = { ...filters };
      memo.result = [];

      return [];
    }

    if (
      (memo.result.length === 0 && families.length > 0) ||
      !isEqual(memo.filters, filters)
    ) {
      memo.filters = { ...filters };

      const { title, categories, sorting } = filters;

      const filteredFonts = families.filter(family => {
        const { category } = fonts[family];

        const titleMatch = family
          .toLowerCase()
          .includes(title.trim().toLowerCase());
        const categoryMatch = categories[category];

        return titleMatch && categoryMatch;
      });

      if (sorting.param === "alpha") {
        const sortedFonts = filteredFonts.sort((a, b) =>
          a.localeCompare(b, {}, { caseFirst: "upper" })
        );

        memo.result =
          sorting.order === "asc" ? sortedFonts : sortedFonts.reverse();
      } else {
        memo.result =
          sorting.order === "asc" ? filteredFonts : filteredFonts.reverse();
      }
    }

    return memo.result;
  };
}
