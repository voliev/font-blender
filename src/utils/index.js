import { ELEMENTS } from "../constants";

export const variantToStyle = variant => {
  const variantElements = variant
    .split(/(?!\d{3})(?=italic|normal)/gi)
    .reverse();
  const styles = variantElements.join(" ").replace("regular", "normal");
  return styles;
};

export const variantsToStyleOptions = variants =>
  variants.reduce(
    (options, variant) => {
      if (/^\d{3}$/gi.test(variant)) {
        variant = `${variant}normal`;
      }
      if (variant === "regular") {
        variant = "400normal";
      }
      if (variant === "italic") {
        variant = "400italic";
      }

      const values = variant.split(/(?!\d{3})(?=italic|normal)/gi);

      if (!options["font-weight"].includes(values[0])) {
        options["font-weight"].push(values[0]);
      }
      if (!options["font-style"].includes(values[1])) {
        options["font-style"].push(values[1]);
      }

      return options;
    },
    { "font-weight": [], "font-style": [] }
  );

export const cssPropertyToCamelCased = prop => {
  const elements = prop.split("-");

  const processedWords = elements.map((element, i) => {
    if (i === 0) return element;

    return element.replace(/\b\w{1}/, match => match.toUpperCase());
  });

  return processedWords.join("");
};

export const isPropTypeHexColor = (props, propName, componentName) => {
  if (!/(^#[0-9A-F]{6}$)|(^#[0-9A-F]{3}$)/i.test(props[propName])) {
    return new Error(
      "Invalid prop `" +
        propName +
        "` supplied to" +
        " `" +
        componentName +
        "`. Validation failed."
    );
  }
};

const generateOptions = selectedFonts =>
  Object.keys(selectedFonts).reduce((options, family) => {
    const { variants } = selectedFonts[family];
    options[family] = variantsToStyleOptions(variants);
    return options;
  }, {});

export const generateElementsStyles = (typeScale, textStyles, fonts) =>
  ELEMENTS.reduce((elementsStyles, element, idx, src) => {
    const { fontFamily } = textStyles[element];

    const exponent = element === "body" ? idx : src.length - idx - 2;

    elementsStyles[element] = generateElementStylesObject(
      element,
      textStyles,
      typeScale,
      fonts,
      fontFamily,
      exponent
    );

    return elementsStyles;
  }, {});

const generateElementStylesObject = (
  element,
  textStyles,
  typeScale,
  fonts,
  family,
  exponent
) => ({
  color: textStyles[element].color,
  fontFamily: fonts[family]
    ? `'${textStyles[element].fontFamily}', '${fonts[family].category}'`
    : family,
  fontSize: `${(
    Math.pow(typeScale.scale, exponent) * typeScale.fontSize
  ).toFixed(2)}px`,
  fontStyle: textStyles[element].fontStyle,
  fontWeight: textStyles[element].fontWeight
});

export const getPropertyOptions = (
  property,
  element,
  textStyles,
  selectedFonts
) => {
  if (property === "font-family") {
    return Object.keys(selectedFonts).sort((a, b) =>
      a.localeCompare(b, {}, { caseFirst: "upper" })
    );
  }

  const options = generateOptions(selectedFonts);
  const { fontFamily } = textStyles[element];

  return options[fontFamily] && options[fontFamily][property];
};

export const getPropertyValue = (textStyles, element, property) => {
  const camelCasedProp = cssPropertyToCamelCased(property);

  return textStyles[element][camelCasedProp];
};

const isSettingsPaneDisabled = state =>
  state.showcase.fontPreview.isVisible || state.showcase.styles.isVisible;

export const isControlDisabled = (state, ...rest) =>
  rest.reduce((acc, curr) => acc || !!curr, isSettingsPaneDisabled(state));

export const getDefaultVariant = variants => {
  // returns empty object for default value
  // of "font-family" property (sans-serif)
  if (!variants) {
    return {};
  }

  if (variants.includes("regular")) {
    return {
      style: "normal",
      weight: 400
    };
  } else if (variants.includes("italic")) {
    return {
      style: "italic",
      weight: 400
    };
  } else {
    return {
      style: variants[0].includes("italic") ? "italic" : "normal",
      weight: parseInt(variants[0])
    };
  }
};

export const numberInputChangeHandler = (
  inputValue,
  defaultValue,
  min,
  max,
  callback
) => {
  const maxDigits = max.toString().length;
  const value = inputValue === "" ? min : inputValue.slice(0, maxDigits);
  const parsedValue =
    value > max
      ? parseInt(value.slice(0, maxDigits - 1), 10)
      : parseInt(value, 10);

  if (isNaN(parsedValue)) {
    callback(defaultValue);
    return;
  }

  if (parsedValue >= min && parsedValue <= max) {
    callback(parsedValue);
    return;
  }

  callback(min);
};
