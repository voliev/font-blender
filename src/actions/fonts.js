import WebFont from "webfontloader";
import { getFonts } from "../reducers/fonts";
import {
  GOOGLE_FONTS_API_ENDPOINT,
  FETCH_FONTS_DATA,
  FETCH_FONTS_DATA_FAILURE,
  FETCH_FONTS_DATA_SUCCESS,
  SET_FONTS,
  UPDATE_FONT
} from "../constants";

export const fetchFontsData = () => ({
  type: FETCH_FONTS_DATA
});

export const fetchFontsDataSuccess = () => ({
  type: FETCH_FONTS_DATA_SUCCESS
});

export const fetchFontsDataFailure = () => ({
  type: FETCH_FONTS_DATA_FAILURE
});

export const setFonts = fonts => ({
  type: SET_FONTS,
  fonts
});

export const startFetchFontsData = () => {
  return async dispatch => {
    const apiRequestURL = `${GOOGLE_FONTS_API_ENDPOINT}?key=${
      process.env.REACT_APP_GOOGLE_FONTS_API_KEY
    }&sort=popularity`;

    dispatch(fetchFontsData());

    try {
      const res = await fetch(apiRequestURL);
      const { items } = await res.json();
      dispatch(fetchFontsDataSuccess());

      const fonts = items.reduce((result, font) => {
        result[font.family] = {
          family: font.family,
          category: font.category,
          variants: font.variants,
          subsets: font.subsets,
          fetching: false,
          active: false,
          requested: false,
          prefetched: false,
          fetched: false
        };

        return result;
      }, {});

      return dispatch(setFonts(fonts));
    } catch (e) {
      dispatch(fetchFontsDataFailure());
      // TODO: Eject and replace service worker with custom one.
      //       Implement fonts' json data caching.
      return dispatch(setFonts({}));
    }
  };
};

export const updateFont = (family, updates) => ({
  type: UPDATE_FONT,
  family,
  updates
});

export const startFetchFont = (family, charRange = undefined) => {
  return (dispatch, getState) => {
    const state = getState();
    const fonts = getFonts(state);
    const font = fonts[family];
    const fetchFull = !charRange;
    const requestString = generateRequestString(font, fetchFull);
    const options = generateRequestOptions(
      requestString,
      charRange,
      dispatch,
      state
    );

    WebFont.load(options);

    const updatedState = getState();
    const updatedFonts = getFonts(updatedState);

    // For "then" chaining in selectedFonts.js async action creator
    return Promise.resolve(updatedFonts[family]);
  };
};

export const generateRequestString = (font, fetchFull = false) => {
  const { family, subsets, variants } = font;

  if (fetchFull) {
    return `${family}:${variants.join(",")}`;
  }

  const weight = variants.includes("regular")
    ? "400"
    : parseInt(variants[0], 10).toString();
  const subset = subsets.includes("latin") ? "latin" : subsets[0];

  return `${family}:${weight}:${subset}`;
};

export const generateRequestOptions = (request, chars, dispatch) => ({
  google: {
    families: [request],
    text: chars
  },
  classes: false,
  timeout: 10000,
  fontloading: getEventListener("fontloading", dispatch, chars),
  fontactive: getEventListener("fontactive", dispatch, chars),
  fontinactive: getEventListener("fontinactive", dispatch, chars)
});

export const getEventListener = (event, dispatch, charRange) => {
  const listeners = {
    fontloading: family => {
      const updates = Object.assign(
        { fetching: true },
        charRange ? { requested: true } : {}
      );
      dispatch(updateFont(family, updates));
    },
    fontactive: family => {
      const updates = Object.assign(
        { active: true, fetching: false },
        charRange ? { prefetched: true } : { fetched: true }
      );
      dispatch(updateFont(family, updates));
    },
    fontinactive: family => {
      const updates = Object.assign(
        { fetching: false },
        charRange ? { active: false } : {}
      );
      dispatch(updateFont(family, updates));
    }
  };

  return listeners[event];
};
