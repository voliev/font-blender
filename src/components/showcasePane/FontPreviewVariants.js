import React from "react";
import PropTypes from "prop-types";
import { SHOWCASE_STRING } from "../../constants";
import { variantToStyle } from "../../utils";
import "./styles/FontPreviewVariants.css";

const FontPreviewVariants = ({ category, family, fontSize, variants }) => (
  <div className="FontPreviewVariants">
    {variants.map(variant => {
      const style = {
        example: `${variantToStyle(
          variant
        )} ${fontSize}px ${family}, ${category}`,
        value: `${variantToStyle(variant)} 16px ${family}, ${category}`
      };

      return (
        <dl className="variant" key={variant}>
          {
            <dt className="variant-value" style={{ font: style.value }}>
              {variant}
            </dt>
          }
          <dd className="variant-example" style={{ font: style.example }}>
            {SHOWCASE_STRING}
          </dd>
        </dl>
      );
    })}
  </div>
);

FontPreviewVariants.propTypes = {
  category: PropTypes.string.isRequired,
  family: PropTypes.string.isRequired,
  fontSize: PropTypes.number.isRequired,
  variants: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default FontPreviewVariants;
