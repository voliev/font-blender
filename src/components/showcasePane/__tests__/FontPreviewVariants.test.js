import React from "react";
import { mount, shallow } from "enzyme";
import FontPreviewVariants from "../FontPreviewVariants";
import { DEFAULT_FONT_SIZE } from "../../../constants";
import { SHOWCASE_STRING } from "../../../constants";
import { variantToStyle } from "../../../utils";
import fonts from "../../../fixtures/fonts";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const font = fonts["Roboto"];
  const props = {
    category: font.category,
    family: font.family,
    fontSize: DEFAULT_FONT_SIZE,
    variants: font.variants,
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<FontPreviewVariants {...props} />)
  };
};

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { category, family, fontSize, variants, wrapper } = setup(mount);

  expect(typeof wrapper.find(FontPreviewVariants).prop("category")).toBe(
    "string"
  );
  expect(wrapper.find(FontPreviewVariants).prop("category")).toBe(category);

  expect(typeof wrapper.find(FontPreviewVariants).prop("family")).toBe(
    "string"
  );
  expect(wrapper.find(FontPreviewVariants).prop("family")).toBe(family);

  expect(typeof wrapper.find(FontPreviewVariants).prop("fontSize")).toBe(
    "number"
  );
  expect(wrapper.find(FontPreviewVariants).prop("fontSize")).toBe(fontSize);

  expect(typeof wrapper.find(FontPreviewVariants).prop("variants")).toBe(
    "object"
  );
  expect(
    wrapper
      .find(FontPreviewVariants)
      .prop("variants")
      .hasOwnProperty("length")
  ).toBeTruthy();
  expect(wrapper.find(FontPreviewVariants).prop("variants")).toEqual(variants);
});

it("should set corresponding styles on font variants", () => {
  const { category, family, fontSize, variants, wrapper } = setup();

  wrapper.find(".variant-value").forEach((node, idx) => {
    expect(node.prop("style")).toEqual({
      font: `${variantToStyle(
        variants[idx]
      )} ${fontSize}px ${family}, ${category}`
    });
  });

  wrapper.find(".variant-example").forEach((node, idx) => {
    expect(node.prop("style")).toEqual({
      font: `${variantToStyle(variants[idx])} 16px ${family}, ${category}`
    });
  });
});

it("should render corresponding variant values", () => {
  const { variants, wrapper } = setup();

  wrapper.find(".variant-value").forEach((node, idx) => {
    expect(node.text()).toBe(variants[idx]);
  });
});

it("should render showcase string", () => {
  const { wrapper } = setup();

  wrapper.find(".variant-example").forEach((node, idx) => {
    expect(node.text()).toBe(SHOWCASE_STRING);
  });
});
