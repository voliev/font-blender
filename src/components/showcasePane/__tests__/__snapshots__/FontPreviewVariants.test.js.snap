// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="FontPreviewVariants"
>
  <dl
    className="variant"
    key="100"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "100 16px Roboto, sans-serif",
        }
      }
    >
      100
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "100 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="100italic"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "italic 100 16px Roboto, sans-serif",
        }
      }
    >
      100italic
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "italic 100 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="300"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "300 16px Roboto, sans-serif",
        }
      }
    >
      300
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "300 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="300italic"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "italic 300 16px Roboto, sans-serif",
        }
      }
    >
      300italic
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "italic 300 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="regular"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "normal 16px Roboto, sans-serif",
        }
      }
    >
      regular
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "normal 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="italic"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "italic 16px Roboto, sans-serif",
        }
      }
    >
      italic
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "italic 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="500"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "500 16px Roboto, sans-serif",
        }
      }
    >
      500
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "500 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="500italic"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "italic 500 16px Roboto, sans-serif",
        }
      }
    >
      500italic
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "italic 500 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="700"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "700 16px Roboto, sans-serif",
        }
      }
    >
      700
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "700 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="700italic"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "italic 700 16px Roboto, sans-serif",
        }
      }
    >
      700italic
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "italic 700 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="900"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "900 16px Roboto, sans-serif",
        }
      }
    >
      900
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "900 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
  <dl
    className="variant"
    key="900italic"
  >
    <dt
      className="variant-value"
      style={
        Object {
          "font": "italic 900 16px Roboto, sans-serif",
        }
      }
    >
      900italic
    </dt>
    <dd
      className="variant-example"
      style={
        Object {
          "font": "italic 900 16px Roboto, sans-serif",
        }
      }
    >
      That's the beauty of music. They can't get that from you...
    </dd>
  </dl>
</div>
`;
